src/main/java/org/radarcns/stream/GeneralStreamGroup.java

```diff
@@ -150,4 +169,25 @@ public void addTopicNames(Collection<String> topicNames) {
         topicList.sort(String.CASE_INSENSITIVE_ORDER);
         return topicList;
     }
+
+    public enum CommitInterval {
+        COMMIT_INTERVAL_FOR_TEN_SECOND(10_000L),
+        COMMIT_INTERVAL_FOR_ONE_MIN(30_000L),
+        COMMIT_INTERVAL_FOR_TEN_MIN(300_000L),
+        COMMIT_INTERVAL_FOR_ONE_HOUR(1800_000L),
+        COMMIT_INTERVAL_FOR_ONE_DAY(86400_000L),
+        COMMIT_INTERVAL_FOR_ONE_WEEK(86400_000L),
```

Comment:
Maximum should probably not exceed 2-3 hours. Otherwise you will need a day of reprocessing whenever you restart the server.